DuckDuckGo/TabViewController.swift

```diff
@@ -1705,20 +1705,15 @@ extension TabViewController: WKNavigationDelegate {
 
     private func loadSpecialErrorPageIfNeeded(error: NSError) {
         guard featureFlagger.isFeatureOn(.sslCertificatesBypass),
-              error.code == NSURLErrorServerCertificateUntrusted,
-              let errorCode = error.userInfo["_kCFStreamErrorCodeKey"] as? Int32,
-              let failedURL = error.failedUrl else {
-            return
-        }
+              error.isServerCertificateUntrusted,
+              let errorType = error.sslErrorType,
+              let failedURL = error.failedUrl else { return }
+
         let tld = storageCache.tld
-        let errorType = SSLErrorType.forErrorCode(Int(errorCode))
         self.failedURL = failedURL
-        errorData = SpecialErrorData(kind: .ssl,
-                                     errorType: errorType.rawValue,
-                                     domain: failedURL.host,
-                                     eTldPlus1: tld.eTLDplus1(failedURL.host))
+        errorData = SpecialErrorData.ssl(type: errorType, domain: failedURL.host!, eTldPlus1: tld.eTLDplus1(failedURL.host))
```

Comment:
any risks in force-unwrapping the host? should we safely unwrap it in the above guard?

Reply:
yeah, sorry, forgot this leftover as it‘s non-optional on macOS, fixed now